lib/spack/spack/repo.py

```diff
@@ -671,43 +671,34 @@ def __init__(self, *repos: "Repo") -> None:
             self.put_last(repo)
 
     @staticmethod
-    def from_paths(
-        *path: str,
+    def from_descriptors(
+        descriptors: "RepoDescriptors",
         cache: spack.util.file_cache.FileCache,
         overrides: Optional[Dict[str, Any]] = None,
     ) -> "RepoPath":
-        repos: List[Repo] = []
+        repo_path, errors = descriptors.construct(cache=cache, fetch=True, overrides=overrides)
 
-        for p in path:
-            try:
-                repos.append(Repo(p, cache=cache, overrides=overrides))
-            except RepoError as e:
-                tty.warn(
-                    f"Failed to initialize repository: '{p}'.",
-                    e.message,
-                    "To remove the bad repository, run this command:",
-                    f"    spack repo rm {p}",
-                )
+        # Merely warn if package repositories from config could not be constructed.
+        if errors:
+            for path, error in errors.items():
+                tty.warn(f"Error constructing repository '{path}': {error}")
```

Comment:
I don't think we want this. If the user configures a repo, we should use it or fail.  If Spack just warns and happily builds from builtin instead of from my private repo, it's really easy for this to silently break CI jobs, private workflows, etc.